Changelog.md

```diff
@@ -55,6 +60,30 @@
 - [#393]: Now `BytesStart::name()` and `BytesEnd::name()` returns `QName`, and
   `BytesStart::local_name()` and `BytesEnd::local_name()` returns `LocalName`
 
+- [#191]: Remove unused `reader.decoder().decode_owned()`. If you ever used it,
+  use `String::from_utf8` instead (which that function did)
+- [#191]: Remove `*_without_bom` methods from the `Attributes` struct because they are useless.
+  Use the same-named methods without that suffix instead. Attribute values cannot contain BOM
+- [#191]: Remove `Reader::decode()` and `Reader::decode_without_bom()`, they was replaced by
+  `Decoder::decode()` and `Decoder::decode_with_bom_removal()`.
+  Use `reader.decoder().decode_*(...)` instead of `reader.decode_*(...)` for now.
+  `Reader::encoding()` was replaced by `Decoder::encoding()` as well
+- [#191]: Remove poorly designed `BytesText::unescape_and_decode_without_bom()` and
+  `BytesText::unescape_and_decode_without_bom_with_custom_entities()`. Although these methods worked
+  as expected, this was only due to good luck. They was replaced by the
+  `BytesStartText::decode_with_bom_removal()`:
+  - conceptually, you should decode BOM only for the first `Text` event from the
+    reader (since now `StartText` event is emitted instead for this)
+  - text before the first tag is not an XML content at all, so it is meaningless
+    to try to unescape something in it
+
+- [#180]: Define the same API for encoding whether feature `encoding` is defined or not.
+  Since then no more replacements does for invalid characters when `encoding` feature
+  is enabled. An error will be returned instead, as for the case when `encoding` is disabled.
```

Comment:
Here you say an error will be returned when `encoding` is disabled, but then below you say the opposite?

Reply:
You mean that:
> If `encoding` feature is disabled, `Cow::Borrowed` is returned instead of `&str`.
?
Yes, probably I should write the types completely. It now returns `Result<Cow<&str>>` (instead of `Result<&str>`), always with `Cow::Borrowed` variant